src/app/activity/bounty/List.ts

```diff
@@ -28,145 +27,131 @@ export const listBounty = async (request: ListRequest): Promise<any> => {
 		break;
-	case 'IN_PROGRESS':
-		dbRecords = bountyCollection.find({ status: BountyStatus.in_progress, 'customerId': request.guildId }).limit(DB_RECORD_LIMIT);
-		break;
-	case 'PAID_BY_ME':
-		dbRecords = bountyCollection.find({ 'createdBy.discordId': listUser.user.id, status: BountyStatus.complete, isIOU: true, 'customerId': request.guildId }).limit(DB_RECORD_LIMIT);
-		IOUList = true;
-		break;
-	case 'UNPAID_BY_ME':
-		dbRecords = bountyCollection.find({ 'createdBy.discordId': listUser.user.id, status: BountyStatus.open, isIOU: true, 'customerId': request.guildId }).limit(DB_RECORD_LIMIT);
-		IOUList = true;
-		break;
-	}
-	if (!(await dbRecords.hasNext())) {
-		return await listUser.send({ content: 'We couldn\'t find any bounties!' });
+	default: 
+		dbRecords = bountyCollection.find({ $or: [ { status: BountyStatus.open } , { status: BountyStatus.in_progress }, { status: BountyStatus.in_review } ], isIOU: { $ne: true }, 'customerId': request.guildId }).sort({ status: -1, createdAt: -1 });
+		listTitle =  "💰 Active Bounties";
 	}
-	return await sendMultipleMessages(listUser, dbRecords, request.guildId, channelName, IOUList);
-};
 
-const sendMultipleMessages = async (listUser: GuildMember, dbRecords: Cursor, guildId: string, bountyChannelName: string, IOUList: boolean): Promise<any> => {
-	if (IOUList) {
-		while (await dbRecords.hasNext()) {
-			const record: Bounty = await dbRecords.next();
-			const messageOptions: MessageEmbedOptions = await generateListEmbedMessage(record, record.paidStatus, guildId);
-			if (record.paidStatus == PaidStatus.unpaid) {
-				messageOptions.footer = {
-					text: '💰 - paid | ❌ - delete ',
-				};
-			}
-			const message: Message = await (listUser.send( { embeds: [messageOptions] } ));
-			if (record.paidStatus == PaidStatus.unpaid) {
-				await message.react('💰');
-				await message.react('❌');
+	const listOfBounties: MessageEmbedOptions = {
+		title: listTitle,
+		url: process.env.BOUNTY_BOARD_URL,
+		color: 1998388,
+		fields: []
+	};
+	let listCount = 0;
+	const bountyList = {};
+	let moreRecords = await dbRecords.hasNext();
+	while (listCount < TOTAL_BOUNTY_LIMIT && moreRecords) {
+		const record: BountyCollection = await dbRecords.next();
+		let cardMessage: Message;
+		if (record.canonicalCard !== undefined) {  
+			// TO DO catch error here and rebuild canonical card if channel or message are missing.
+			try {
+				cardMessage = await DiscordUtils.getMessagefromMessageId(record.canonicalCard.messageId, await DiscordUtils.getTextChannelfromChannelId(record.canonicalCard.channelId));
+			} catch {
+				cardMessage = undefined;
 			}
 		}
-	} else {
-		const listOfBounties = [];
-		while (listOfBounties.length < 10 && await dbRecords.hasNext()) {
-			const record: Bounty = await dbRecords.next();
-			const messageOptions: MessageEmbedOptions = await generateListEmbedMessage(record, record.status, guildId);
-			listOfBounties.push(messageOptions);
+		if (!bountyList[record.status]) {
```

Comment:
nit: I think this initialization can happen by default outside of the loop

Reply:
I don't think so, because we need to initialize it for whatever status we run into with the current record. We don't know what statuses we will hit outside the loop.

Reply:
You're right -  thank you for the explanation